### PR TITLE
Long-long pull request 

### PR DESCRIPTION
**Pull request by Varkh**

### CODE REVIEW

File: modules/dbSchema/topic.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+var mongoose = require('mongoose');
+mongoose.Promise = global.Promise;
+var fs = require('fs');
+var logger = require('.././logger').logger;
+var CommentSchema=require('./comment');
+
+var TopicSchema = new mongoose.Schema({
+    name: {
```
Comment (Varkh) [inline modules/dbSchema/topic.js:8]: @kds1660 How would you handle topics with same name?
Just a question

File: modules/dbSchema/user.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        unique: true,
+        required: true
+    },
+    password: {
+        type: String,
+        required: true
+    }
+});
+UserSchema.methods.validPassword = function( pwd ) {
+    return ( this.password === pwd );
```
Comment (Varkh) [inline modules/dbSchema/user.js]: @kds1660 Read about hashing passwords

File: public/controllers/rootController.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+app.controller('mainController', function($scope,$timeout,requestService){
+    $scope.isLogged=0;
+    $scope.alert={};
+    $scope.alert.text='s';
+    $scope.setName = function(name){
+        $scope.isLogged = name;
+    };
+    $scope.setAllert=function (trueFalse,text) {
+        console.log(trueFalse,text)
+        if (trueFalse===true) {
```
Comment (Varkh) [inline public/controllers/rootController.js]: @kds1660 copy-paste?

File: public/controllers/loginController.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+app.controller('loginController',function($scope,$timeout,requestService,buttonService){
```
Comment (Varkh) [inline public/controllers/loginController.js]: Create service, move login logic to service. Leave button logic in Controller.

File: public/directives/directives.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+app.directive("topicPost", function(buttonService) {
+    return {
+        restrict : "E",
+        templateUrl : "templates/topicTemplate.html",
+    };
+});
+
+app.directive("loginLogout",function(buttonService) {
```
Comment (Varkh) [inline public/directives/directives.js]: @kds1660 1 file, 1 directive

File: public/index.html [hunk 6]
```diff
@@ -1,45 +1,100 @@
...
-    <script src="/bower_components/bootstrap/dist/js/bootstrap.js"></script>
+
+    <title>Blog Post - Start Bootstrap Template</title>
     <script src="/bower_components/tinymce/tinymce.js"></script>
-    <script src="buttons.js"></script>
-    <script src="engine.js"></script>
+    <script src="/bower_components/angular/angular.js"></script>
+    <script src="mainApp.js" defer></script>
+
+    <script src="services/services.js" defer></script>
```
Comment (Varkh) [inline public/index.html]: @kds1660 I was expecting Gulp usage from you.

File: public/scroll.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                template = $('#mustache_topic').html();
+                data={
+                    name:response1[0][index].name,
+                    date: new Date(response1[0][index].date).toLocaleString('en-GB',options),
+                    author:response1[0][index].author,
+                    text:response1[0][index].text,
+                    img:response1[0][index].img,
+                    comments:response1[0][index].comments
+                };
+                var $copy = Mustache.to_html(template, data);
```
Comment (Varkh) [inline public/scroll.js]: @kds1660 костыли-костыли
time to rewrite jquery and mustache to angular

File: modules/dbSchema/topic.js (hunk 1 above)
Comment (kds1660) [reply modules/dbSchema/topic.js:8]: i cant add topic with existing name, name set in schema as unique

File: modules/dbSchema/user.js (hunk 2 above)
Comment (kds1660) [reply modules/dbSchema/user.js]: this is thing, that i want to do (and passport checking for all requests) but dont do that yet

File: public/controllers/rootController.js (hunk 3 above)
Comment (kds1660) [reply public/controllers/rootController.js]: oops

File: public/controllers/loginController.js (hunk 4 above)
Comment (kds1660) [reply public/controllers/loginController.js]: something dont understand - where is login logic and button logic, almost each button = request with action on response, I planned remove all button logic as buttonclick= function () {service.method}

**Pull request merged**